### PR TITLE
Import default template args

### LINKED ISSUE #310: template parameter default values are not imported
Example code:
```
template <typename T=int> void declToImport(T &t) { };
```
After import the `=int` default value is missing.

### PR DESCRIPTION
**Pull request by martong**
Fixes #310

### CODE REVIEW

File: lib/AST/ASTImporter.cpp [hunk 1]
```diff
@@ -4174,7 +4174,7 @@ Decl *ASTNodeImporter::VisitTemplateTypeParmDecl(TemplateTypeParmDecl *D) {
   // is created.
   
   // FIXME: Import default argument.
-  return TemplateTypeParmDecl::Create(Importer.getToContext(),
+  auto ToD = TemplateTypeParmDecl::Create(Importer.getToContext(),
```
Comment (Xazax-hun) [inline lib/AST/ASTImporter.cpp]: Fixme above is no longer relevant.

**Review by Xazax-hun — COMMENTED** [review]
What about value templates with default arguments?

**Comment by martong** [comment]
Added new test for this case:
```
      template<typename T0, typename T1 = int> class A;
      template<typename T0 = int, typename T1> class A;
```

Added new test for VarTemplateDecl.

**Review by balazske — COMMENTED** [review]
It looks good, but only type kind of template arguments are used at the test. Does it work similarly with value and template type of arguments?

**Comment by gamesh411** [comment]
Can one of the admins verify this patch?

**Comment by martong** [comment]
Not important for a while, because all specializations/instantiations have the concrete parameter. We did not see any error caused by this.

**Pull request closed without merging**